### PR TITLE
upgrade cranelift, wasi-common, witx deps, and friends

### PR DESCRIPTION
**Pull request by pchickey**
* pull in latest Cranelift release 0.51
* upgrade to wasi-common 0.7, which fixes a lot of the WasiCtxBuilder
  stuff that required many extra unwraps
* pull in latest WASI repo. This moves the preview0 stuff to another
  directory. Biggest change here is that the lucetc deb, which includes
  the wasi witx files, has a slightly different file layout now.
* transitive upgrades to target-lexicon, faerie

### CODE REVIEW

File: lucet-validate/Cargo.toml [hunk 1]
```diff
@@ -39,6 +39,6 @@ assets = [
     ["LICENSE", "/opt/fst-lucet-validate/share/doc/lucet-validate/", "644"],
     ["../wasi/phases/unstable/witx/typenames.witx",
      "/opt/fst-lucet-validate/share/wasi/unstable/typenames.witx", "644"],
-    ["../wasi/phases/unstable/witx/wasi_unstable_preview0.witx",
+    ["../wasi/phases/unstable/witx/wasi_unstable.witx",
      "/opt/fst-lucet-validate/share/wasi/unstable/wasi_unstable_preview0.witx", "644"],
```
Comment (acfoltzer) [inline lucet-validate/Cargo.toml]: Do we also want to change the destination filename?

File: lucet-wasi-fuzz/src/main.rs [hunk 2]
```diff
@@ -368,14 +368,11 @@ fn run_with_stdout<P: AsRef<Path>>(
     tmpdir: &TempDir,
     path: P,
 ) -> Result<(__wasi_exitcode_t, Vec<u8>), Error> {
-    let ctx = WasiCtxBuilder::new()
-        .map_err(|e| format_err!("WasiCtxBuilder: {}", e))?
-        .args(["gen"].iter())
-        .map_err(|e| format_err!("WasiCtxBuilder args: {}", e))?;
+    let ctx = WasiCtxBuilder::new().args(&["gen"]);
```
Comment (acfoltzer) [inline lucet-wasi-fuzz/src/main.rs]: :pray:

File: lucet-wasi/src/main.rs [hunk 3]
```diff
@@ -235,13 +235,9 @@ fn run(config: Config<'_>) {
...
             .collect::<Vec<&str>>();
         let mut ctx = WasiCtxBuilder::new()
-            .expect("wasi context can be built")
             .args(args.iter())
-            .expect("arguments can be stored")
             .inherit_stdio()
-            .expect("stdio can be inherited")
-            .inherit_env()
-            .expect("environment can be inherited");
+            .inherit_env();
```
Comment (acfoltzer) [inline lucet-wasi/src/main.rs]: :pray: :pray:

File: lucet-validate/Cargo.toml (hunk 1 above)
Comment (pchickey) [reply lucet-validate/Cargo.toml]: Ah, good catch

Comment (pchickey) [reply lucet-validate/Cargo.toml]: These paths are wrong, we apparently never make the debs in CI. I'll add it to the makefile.

**Comment by pchickey** [comment]
I added testing of the deb creation to `make test`. This will increase Dockerfile build times, and CI run times. Hopefully GH Actions gets docker image caching soon and it doesn't hurt quite as badly, but I'd rather have this stuff tested than find out downstream.

**Pull request merged**